build/common.ps1

```diff
@@ -454,20 +454,44 @@ Function Test-CoreProjects {
     $xtests | Test-XProject -Configuration $Configuration
 }
 
+Function Test-MSBuildVersionPresent {
+    [CmdletBinding()]
+    param(
+        [parameter(Mandatory=$True, position=0)]
+        [string]$MSBuildVersion
+    )
+    
+   	$MSBuildExe = Get-MSBuildExe $MSBuildVersion
+
+    Test-Path $MSBuildExe
+}
+
+Function Get-MSBuildExe {
+    param(
+        [parameter(Mandatory=$True, position=0)]
+        [string]$MSBuildVersion
+    )
+    
+    $MSBuildExe = Join-Path $MSBuildRoot ($MSBuildVersion + ".0")
+    Join-Path $MSBuildExe $MSBuildExeRelPath
+}
+
 Function Build-ClientsProjects {
     [CmdletBinding()]
     param(
         [string]$Configuration = $DefaultConfiguration,
         [string]$ReleaseLabel = $DefaultReleaseLabel,
         [int]$BuildNumber = (Get-BuildNumber),
+        [parameter(Mandatory=$True)]
+        [string]$MSBuildVersion,
```

Comment:
I considered this but didn't want to allow fallback behavior on something so crucial as a build version, especially when there's no bias toward one version over another. 14 is no better than 15. If the parameter is unpassed, then something has broken something.